pl_examples/utils/loaders.py

```diff
@@ -8,28 +8,37 @@
 from torch.utils.data import DataLoader
 
 
-class AsynchronousLoader(DataLoader):
+class AsynchronousLoader(object):
     """
     Class for asynchronously loading from CPU memory to device memory
 
     Parameters
     ----------
     dataset: PyTorch Dataset
-        The PyTorch dataset we're loading
+        The PyTorch dataset we're loading.
+        Exactly one of dataset or dataloader must be specified
+        This must also be finite and of constant length
+    dataloader: PyTorch DataLoader
+        The PyTorch DataLoader we're using to load.
+        Exactly one of dataset or dataloader must be specified
+        This must also be finite and of constant length
     device: PyTorch Device
         The PyTorch device we are loading to
-    num_workers: Integer
-        Number of worker processes to use for loading from storage and collating the batches in CPU memory
-    queue_size: Integer
+    q_size: Integer
         Size of the queue used to store the data loaded to the device
+    **kwargs:
+        Any additional arguments to pass to the dataloader if we're constructing one here
     """
 
-    def __init__(self, dataset, device, num_workers=8, queue_size=10, **kwargs):
-        super(AsynchronousLoader, self).__init__(dataset=dataset,
-                                                 pin_memory=True,
-                                                 num_workers=num_workers,
-                                                 **kwargs
-                                                 )
+    def __init__(self, dataset=None, dataloader=None, device=torch.device('cuda', 0), q_size=10, **kwargs):
```

Comment:
One more question: Can we maybe change the order of dataset and data loader here? I think by default the user specifies a loader and just wants it to be wrapped. Therefore it should also be the first positional argument here.

Also I'm thinking if we can maybe check for types with a single argument here.
Something like 
```python
if is instance(data, torch.utils.data.Dataset):
    data = ... # Instatiate loader
# otherwise assume data is already a loader providing an `__iter__` and `__next__` interface
```

Reply:
Will do, make sense

Reply:
@justusschock done!